packages/relay-compiler/codegen/RelayCodeGenerator.js

```diff
@@ -66,6 +66,13 @@ const RelayCodeGenVisitor = {
         node.metadata.idType,
       );
     }
+    if (node.kind === 'Fragment' || node.kind === 'LinkedField') {
+      const selections = flattenArray(node.selections);
+      const idFieldSelection = selections.find(
+        selection => selection.metadata && selection.metadata.isDataID,
+      );
+      node.idField = idFieldSelection && idFieldSelection.name;
```

Comment:
since this gets toJSON'd we probably want to make sure it's always `null` if there is no idFieldSelection, so that all Fragments and LinkedFields have the same property set and same hidden class

Reply:
Also, this logic should probably go in a transform and have its own test, not the code generator